### ISSUE TITLE
inconsistent nested form in Phoenix.Html.Form

### ISSUE THREAD

**Issue by totaltrash** [ash-project/ash_phoenix#42]
**Describe the bug**

when inspecting the Phoenix.Html.Form
```elixir
{inspect form.source.forms[:nested] }
```
if I haven't added any forms with add_form, it gives `[]`
but, if i add_form, then remove form, it is `nil`

**Expected behavior**

** Runtime
 - Elixir version
 - Erlang version
 - OS
 - Ash version: main
 - any related extension versions

**Additional context**

**Comment by zachdaniel** [comment]
I think this may actually stem from a slighly incorrect usage of add forms, i.e `remove_form(form, [:nested])` where `[:nested]` is a `:list` type nested form, meaning to remove a form there you should actually pass an integer. However, I'll handle that case and make that logic remove *all* forms.

**Comment by totaltrash** [comment]
This is what i'm passing to remove_form, to remove the last item from the form:

"project[staff][0]"

which results in the nil form, not empty list

**Comment by zachdaniel** [comment]
Is it still
Happening after the latest update?

**Comment by totaltrash** [comment]
Yes, still happening

**Comment by zachdaniel** [comment]
Very surprising, looking at how `remove_form` does its work when given an integer...

```elixir
    new_forms =
      form.forms
      |> Map.put_new(key, [])
      |> Map.update!(key, fn forms ->
        forms
        |> Kernel.||([])
        |> List.delete_at(i)
        |> Enum.with_index()
        |> Enum.map(fn {nested_form, i} ->
          %{nested_form | name: form.name <> "[#{key}][#{i}]", id: form.id <> "_#{key}_#{i}"}
        end)
      end)

    %{
      form
      | forms: new_forms,
        any_removed?: form.any_removed? || any_removed?,
        touched_forms: MapSet.put(form.touched_forms, to_string(key)),
        form_keys: new_config,
        opts: Keyword.put(form.opts, :forms, new_config)
    }
```

**Comment by zachdaniel** [comment]
Reading that code it looks like we should be setting the key's value to a list effectively unconditionally.

**Comment by zachdaniel** [comment]
Oh, I wonder, are you passing `validate?: false` into it? If you aren't then `nil` in forms may actually make sense...

**Comment by totaltrash** [comment]
ah, ok, so confirmed, when `validate?: false`, I'm left with an empty list, when validate, it's nil. So it sounds like i need to validate the form on create to get a consistent behaviour (always nil if no nested forms)?

**Comment by totaltrash** [comment]
actually, no that doesn't work. it's still an empty list if i validate on create (assuming validating with an empty params map would actually do any validation?). 

Feels a bit weird, but i can work around it with a wrapper perhaps

**Comment by totaltrash** [comment]
Oh, this fixes it (after creating the form): `form = AshPhoenix.Form.validate(form, AshPhoenix.Form.params(form), errors: false)`

**Comment by totaltrash** [comment]
Yeah, so I think in the same way that we now include a `validate` in `add_form` and `remove_form`, it feels like we need to include a validate on form create, if that form contains nested forms that could be added later (if we want to experience the same behaviour wrt no nested forms currently added)

**Comment by zachdaniel** [comment]
Yeah, so I think it may just not be a guarantee that we necessarily need to make, as long as it behaves properly when adding/submitting. So really it seems like you just need a function in `AshPhoenix` to give you an API for what you're doing, and it can be Ash's problem that the form there may be an empty list or nil at some point internally.

**Comment by zachdaniel** [comment]
Open to ideas for that interface function, i.e `get_forms/2` or `has_form?` accepting a non-integer last path and checking if the list is empty or nil.

**Comment by totaltrash** [comment]
Yes, sounds good. I thought i might have been digging into the internals.

Would `get_forms/2` only apply to nests `:list` type forms? If so, `get_forms/2` that (when empty) returns an empty list sounds good. If it can work for :single type forms too, then you would expect that to return nil, although I'm not sure that is needs to work for singles, as we have `get_form` already?

**Comment by zachdaniel** [comment]
I think we could make it work for single forms, we just want it to have a consistent return type, i.e *always* a list.

`has_form?/2` exists already that will tell you wether or not there is a form at a given path and it works for lists, singles, and also just "are there any forms at a list path", maybe that actually does what you want come to think of it?
